itest/lnd_misc_test.go

```diff
@@ -876,52 +865,132 @@ func testSweepAllCoins(ht *lntest.HarnessTest) {
 		require.NoError(ht, err, "timeout assertTxLabel")
 	}
 
-	sweepTxStr := sweepTx.TxHash().String()
-	waitTxLabel(sweepTxStr, sendCoinsLabel)
-
-	// While we are looking at labels, we test our label transaction
-	// command to make sure it is behaving as expected. First, we try to
-	// label our transaction with an empty label, and check that we fail as
-	// expected.
-	sweepHash := sweepTx.TxHash()
-	err := ainz.RPC.LabelTransactionAssertErr(
-		&walletrpc.LabelTransactionRequest{
-			Txid:      sweepHash[:],
-			Label:     "",
-			Overwrite: false,
-		},
+	// List unspent utxos so that we can fetch outpoints for the test below.
+	res := ainz.RPC.ListUnspent(
+		&walletrpc.ListUnspentRequest{},
 	)
 
-	// Our error will be wrapped in a rpc error, so we check that it
-	// contains the error we expect.
-	errZeroLabel := "cannot label transaction with empty label"
-	require.Contains(ht, err.Error(), errZeroLabel)
-
-	// Next, we try to relabel our transaction without setting the overwrite
-	// boolean. We expect this to fail, because the wallet requires setting
-	// of this param to prevent accidental overwrite of labels.
-	err = ainz.RPC.LabelTransactionAssertErr(
-		&walletrpc.LabelTransactionRequest{
-			Txid:      sweepHash[:],
-			Label:     "label that will not work",
-			Overwrite: false,
+	tcs := []struct {
```

Comment:
I've laid out my view on table-driven test [here](https://github.com/lightningnetwork/lnd/pull/8800#discussion_r1634047064) and I think we should avoid it in itest.

Reply:
In general, are you saying that if a function has more than two args we should not use TDD for it?
>I like using table-driven tests when the outputs of a function are only determined by one or two inputs in a test setup,

Reply:
It depends - but we should discourage using it in itest because it's difficult to maintain.

Reply:
I am sorry I am having difficulty refactoring this to not use TDD. I am not sure what the non TDD version should look like. Would it be something like this?

```go

similarfunction := func(t *testing.T, req *sendcoinsRequest, someconditionalsthatwecheck){
----
}

// With select utxos
ht.Run(){
similarfunction()
}

// With no select utxos
ht.Run(){
similarfunction()
}

```

Would I need to refactor `testsendcoinsselectutxos` as well?